app/parse.py

```diff
@@ -8,8 +15,46 @@ class Quote:
     tags: list[str]
 
 
+def get_single_quotes(quote_soup: Tag) -> Quote:
+    return Quote(
+        text=quote_soup.select_one(".text").text,
+        author=quote_soup.select_one(".author").text,
+        tags=[tag.text for tag in quote_soup.select(".tag")]
+    )
+
+
+def get_single_page_quotes(page_soup: BeautifulSoup) -> list[Quote]:
+    quotes = page_soup.select(".quote")
+    return [get_single_quotes(quote_soup) for quote_soup in quotes]
+
+
+def parse_quotes(url: str) -> list[Quote]:
+    all_quotes = []
+    while url:
+        print(f"Scraping {url}")
+        page = requests.get(url).content
+        soup = BeautifulSoup(page, "html.parser")
+        all_quotes.extend(get_single_page_quotes(soup))
+        next_page = soup.select_one(".next > a")
+        url = BASE_URL + next_page["href"] if next_page else None
```

Comment:
When concatenating URLs, using urljoin from urllib.parse is generally more robust than string concatenation, especially when dealing with URLs.